packages/repository/src/decorators/repository.ts

```diff
@@ -4,15 +4,23 @@
 // License text available at https://opensource.org/licenses/MIT
 
 import * as assert from 'assert';
-import {Model, Entity} from '../model';
-import {Repository} from '../repository';
+import {Entity} from '../model';
 import {DataSource} from '../datasource';
 import {
   DefaultCrudRepository,
   DataSourceConstructor,
 } from '../legacy-juggler-bridge';
 import {juggler} from '../loopback-datasource-juggler';
-import {inject, Context, Injection} from '@loopback/context';
+import {inject, Context, Injection, BoundValue} from '@loopback/context';
+
+/**
+ * Type definition for decorators returned by `@repository` decorator factory
+ */
+export type RepositoryDecorator = (
+  target: Object,
+  key?: string | symbol,
+  descriptorOrIndex?: TypedPropertyDescriptor<BoundValue> | number,
```

Comment:
Any reason we can't return the type as `TypedPropertyDescriptor<Repository<T>>`?

Reply:
^^^ same as reasoning above: to get rid of generics

Reply:
Because github hid my comments, I'm copypasting from above comment:

Honestly, `descriptor` argument is pretty irrelevant considering we're never going to be using it as a propertyDescriptor since `@repository` has no use cases for decorating methods themselves (and if a user tries it, it errors out). My reasoning for replacing `Repository<T>` was to get rid of the usage of generics since we don't need it at all (at least from what I understand).

I might change the argument name `descriptor` to something else so that it's less confusing.

Reply:
Maybe I should put this PR on hold for the author of #744 to hear his inputs on taking in `Class.name` as the argument